### PR TITLE
s22-4pm-3 kz - New leaderboard button on play page

### LINKED ISSUE #13: On the play page, add a button that goes to the leaderboard for the commons. For this issue , the leaderboard can be a placeholder page.

### PR DESCRIPTION
**Pull request by KelvinZhang123**
# Overview

Added files are titled as LeaderboardPlayPageButton to avoid confusion with other filed related to the leaderboard. LeaderboardPlayPageButton indicates that the files are related to the leaderboard button on the play page.

Added a new column to the table on the play page titled "Leaderboard" with a new leaderboard png icon as well as a button titled "Board" that the user can click which should direct them to the leaderboard page. 

Currently there is a placeholder link and clicking the button will only direct the user to '../leaderboard/' but once the leaderboard page is created the proper link can be added. On localhost, clicking the button will direct you to 'http://localhost:8080/leaderboard/'.

Attached below is what the new play page looks like on localhost:
Screen Shot 2022-06-01 at 5 58 01 PM

# Storybook additions
Once merged the component should also appear on storybook and should look like what is displayed below:
Screen Shot 2022-06-01 at 5 59 46 PM

### CODE REVIEW

File: frontend/src/main/components/Commons/LeaderboardPlayPageButton.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                <Row>
+                    <Col>
+                        <Card.Text>
+                            <img alt="Leaderoard Icon" width={100} height ={100} className="icon" src={leaderboardIcon}></img>
+                        </Card.Text>
+                    </Col>
+                    <Col>
+                        <div>
+                        <Link to='/leaderboard/'>
+                        <Button variant="outline-danger" data-testid={"leaderboard-playpage-button"}>Board</Button>
```
Comment (peterbrede) [inline frontend/src/main/components/Commons/LeaderboardPlayPageButton.js:24]: remove variant="outline-danger", maybe change name of button to Leaderboard
```suggestion
                        <Button data-testid={"leaderboard-playpage-button"}>Leaderboard</Button>
```

**Review by calvitronic — DISMISSED** [review]
This is a very well-executed PR and the changes seem to correctly implement features according to the issue's requirements. No commented out code from what I can see, LGTM!

**Review by peterbrede — CHANGES_REQUESTED** [review]

**Review by peterbrede — APPROVED** [review]

**Review by peterbrede — CHANGES_REQUESTED** [review]

**Review by peterbrede — DISMISSED** [review]

**Review by calvitronic — DISMISSED** [review]
Good after those new changes, LGTM!

File: frontend/src/main/components/Commons/LeaderboardPlayPageButton.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import {Link} from 'react-router-dom'
+
+const Leaderboard = () =>  {
+    
+    return (
+        <Card>
+        <Card.Header as="h5">Leaderboard</Card.Header>
+        <Card.Body>
+            {}
+
```
Comment (andrewhlu) [inline frontend/src/main/components/Commons/LeaderboardPlayPageButton.js:13]: is there a reason why there's a set of curly braces here?

File: frontend/src/main/components/Commons/LeaderboardPlayPageButton.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        <Card>
+        <Card.Header as="h5">Leaderboard</Card.Header>
+        <Card.Body>
+            {}
+            
+            <Card.Title></Card.Title>
+                <Row>
+                    <Col>
+                        <Card.Text>
+                            <img alt="Leaderoard Icon" width={100} height ={100} className="icon" src={leaderboardIcon}></img>
```
Comment (andrewhlu) [inline frontend/src/main/components/Commons/LeaderboardPlayPageButton.js:18]: a nitpick change to remove a space.

```suggestion
                            <img alt="Leaderoard Icon" width={100} height={100} className="icon" src={leaderboardIcon}></img>
```

File: frontend/src/tests/components/Commons/LeaderboardPlayPageButton.test.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                <MemoryRouter>
+                    <LeaderboardPlayPage />
+                </MemoryRouter>
+            </QueryClientProvider>
+        );
+
+        expect(await screen.findByText("Board")).toBeInTheDocument();
+        const PlayPageMenu = screen.getByTestId("leaderboard-playpage-button");
+        expect(PlayPageMenu).toBeInTheDocument();
+
```
Comment (andrewhlu) [inline frontend/src/tests/components/Commons/LeaderboardPlayPageButton.test.js:31]: I'd suggest also adding some statements to check that, if the button is clicked, we get navigated to the correct page.

**Review by andrewhlu — DISMISSED** [review]

File: frontend/src/main/components/Commons/LeaderboardPlayPageButton.js (hunk 2 above)
Comment (calvitronic) [reply frontend/src/main/components/Commons/LeaderboardPlayPageButton.js]: No need, deleted.

File: frontend/src/tests/components/Commons/LeaderboardPlayPageButton.test.js (hunk 4 above)
Comment (calvitronic) [reply frontend/src/tests/components/Commons/LeaderboardPlayPageButton.test.js:31]: Working on this now.

Comment (calvitronic) [reply frontend/src/tests/components/Commons/LeaderboardPlayPageButton.test.js:31]: Does this do the trick?

**Review by KelvinZhang123 — COMMENTED** [review]
These changes look great. Thanks Calvin!

**Review by calvitronic — DISMISSED** [review]
Updated tests to check if button redirects correctly, updated link to redirect to correct location as well, LGTM.

**Review by pconrad — APPROVED** [review]
LGTM!

**Review by pconrad — CHANGES_REQUESTED** [review]
I tried deploying this on QA, and ran into some bugs when visiting the play page for a new commons.  This was after resetting the database and redeploying.

Let me encourage you to test on
https://dashboard.heroku.com/apps/s22-4pm-3-happycows-qa with a sequence of
* Create new commons
* Join new commons as admin user
* Visit play page as admin user
* Logout, and login as non-admin user
* Join new commons
* Visit play page as non-admin user


And see if you can get it to be stable.
Let me know when I should take a new look.

**Review by calvitronic — APPROVED** [review]
Seemed to fix unstable behavior by adding optional chaining to commons.id reference.

**Comment by pconrad** [comment]
Closed; 6pm codebase already has a similar feature.

**Pull request closed without merging**